src/__tests__/withProps.test.js

```diff
@@ -39,4 +39,31 @@ describe('withProps', () => {
     )
     expect(wrapper.equals(<div />)).toBeTruthy()
   })
+
+  it("doesn't compact surrounded HOC that hoists statics", () => {
+    const hoc = BaseComponent => props => <BaseComponent {...props} foo="bar" />
+    const StaticHoistingHOC = hoistStatics(hoc)
+
+    const Component = compose(withProps({}), StaticHoistingHOC, withProps({}))(
+      'div',
+    )
+
+    const wrapper = shallow(<Component />)
+    expect(wrapper.prop('foo')).toBe('bar')
+  })
+
+  it("doesn't compact double-surrounded HOC that hoists statics", () => {
```

Comment:
This tests `isMapperComponent()` (test fails against `master` version of `utils/createHOCFromMapper`